src/communication/bolt/v1/session.hpp

```diff
@@ -120,6 +120,9 @@ class Session {
         return;
       }
       handshake_done_ = true;
+      // Update the decoder's Bolt version (v5 has changed the undelying structure)
+      decoder_.UpdateVersion(version_.major);
+      encoder_.UpdateVersion(version_.major);
```

Comment:
Yes, we do not know which version of the protocol we will end up using until the server and client negotiate it. 
We could also destroy the old de/encoder and create a new one that takes in the version, but I wasn't sure if this would have unintended consequences (on the buffer or some other hidden state), so I just added the function call.
I'll add a comment to the code with some ideas about how to improve this part later on.